lib/matplotlib/backend_tools.py

```diff
@@ -0,0 +1,882 @@
+"""
+Abstract base classes define the primitives for Tools.
+These tools are used by `NavigationBase`
+
+:class:`ToolBase`
+    Simple stateless tool
+
+:class:`ToolToggleBase`
+    Tool that has two states, only one Toggle tool can be
+    active at any given time for the same `Navigation`
+"""
+
+
+from matplotlib import rcParams
+from matplotlib._pylab_helpers import Gcf
+import matplotlib.cbook as cbook
+from weakref import WeakKeyDictionary
+import numpy as np
+
+
+class Cursors:
+    """Simple namespace for cursor reference"""
+    HAND, POINTER, SELECT_REGION, MOVE = list(range(4))
+cursors = Cursors()
+
+
+class ToolBase(object):
+    """Base tool class
+
+    A base tool, only implements `trigger` method or not method at all.
+    The tool is instantiated by `matplotlib.backend_bases.NavigationBase`
+
+    Attributes
+    ----------
+    navigation: `matplotlib.backend_bases.NavigationBase`
+        Navigation that controls this Tool
+    figure: `FigureCanvas`
+        Figure instance that is affected by this Tool
+    name: String
+        Used as **Id** of the tool, has to be unique among tools of the same
+        Navigation
+    """
+
+    keymap = None
```

Comment:
How about a compromise?  Renaming this attribute as ``default_keymap`` gives us more flexibility, allowing us to override this default during tool creation.  This also means that user can prevent tools from overwriting keymaps of belonging to other tools if a collision occurs.